src/Domain/DonationRepository.php

```diff
@@ -283,9 +247,12 @@ public function findWithExpiredMatching(\DateTimeImmutable $now): array
         // `ExpireMatchFunds`, it makes more sense to opt it out of result caching
         // here rather than take the performance hit of a full query cache clear
         // after every single persisted donation.
-        return $qb->getQuery()
+        /** @var list<array{uuid: UuidInterface}> $rows */
+        $rows = $qb->getQuery()
             ->disableResultCache()
```

Comment:
I think this existing line and the comment above are misleading, as I since learnt we're probably not using the result cache in general. But I'm in two minds about removing it because we have a ticket to probably change that soon-ish.

Reply:
Right, I think I'll merge as-is then, especially since this PR is already a big diff.